brig/cmd/brig/commands/project_create.go

```diff
@@ -0,0 +1,537 @@
+}
+
+var projectCreate = &cobra.Command{
+	Use:   "create",
+	Short: "create a new Brigade project",
+	Long:  projectCreateUsage,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if projectCreateFromProject != "" && projectCreateConfig != "" {
+			return errors.New("cannot specify both -f/--config and -p/--from-project.")
+		}
+		return createProject(cmd.OutOrStderr())
+	},
+}
+
+func createProject(out io.Writer) error {
+	c, err := kubeClient()
+	if err != nil {
+		return err
+	}
+
+	store := kube.New(c, globalNamespace)
+
+	proj := defaultProject
+	if projectCreateConfig != "" {
+		if proj, err = loadProjectConfig(projectCreateConfig, proj); err != nil {
+			return err
+		}
+	}
+
+	if projectCreateFromProject != "" {
+		var err error
+		proj, err = store.GetProject(projectCreateFromProject)
+		if err != nil {
+			return fmt.Errorf("could not load %s: %s", projectCreateFromProject, err)
+		}
+	}
+
+	if !projectCreateNoPrompts {
+		if err := projectCreatePrompts(proj); err != nil {
+			return err
+		}
+	}
+
+	secret, err := kube.SecretFromProject(proj)
+	if err != nil {
+		return err
+	}
+	data, err := json.MarshalIndent(secret, "", "  ")
+	if err != nil {
+		return err
+	}
+
+	if globalVerbose {
+		out.Write(data)
+		fmt.Println()
+	}
+
+	if projectCreateOut != "" {
+		if err := ioutil.WriteFile(projectCreateOut, data, 0755); err != nil {
+			return err
+		}
+	}
+
+	fmt.Fprintf(out, "Project ID: %s\n", proj.ID)
+	if projectCreateDryRun {
+		return nil
+	}
+
+	if !projectCreateReplace {
+		if _, err = store.GetProject(proj.Name); err == nil {
+			return fmt.Errorf("Project %s already exists. Refusing to overwrite.", proj.Name)
+		}
+	}
+
+	// Store the project
+	if err := store.CreateProject(proj); err != nil {
+		return err
+	}
+	return nil
+}
+
+// projectCreatePrompts handles all of the prompts.
+//
+// Default values are read from the given project. Values are then
+// replaced on that object.
+func projectCreatePrompts(p *brigade.Project) error {
+	if err := survey.AskOne(&survey.Input{
+		Message: "Project name",
+		Help:    "By convention, this is user/project or org/project",
+		Default: p.Name,
+	}, &p.Name, survey.Required); err != nil {
+		return err
+	}
+
+	qs := []*survey.Question{
+		{
+			Name: "name",
+			Prompt: &survey.Input{
+				Message: "Full repository name",
+				Help:    "A protocol-neutral path to your repo",
+				Default: p.Repo.Name,
+			},
+		},
+		{
+			Name: "cloneURL",
+			Prompt: &survey.Input{
+				Message: "Clone URL (https://github.com/your/repo.git)",
+				Help:    "The URL that Git should use to clone. The protocol (https, git, ssh) will determine how the repo is fetched.",
+				Default: p.Repo.CloneURL,
+			},
+		},
+	}
+
+	abort := "project could not be saved: %s"
+	if err := survey.Ask(qs, &p.Repo); err != nil {
+		return fmt.Errorf(abort, err)
+	}
+
+	// Only prompt for key if clone URL requires it
+	if strings.HasPrefix(p.Repo.CloneURL, "ssh") || strings.HasPrefix(p.Repo.CloneURL, "git+ssh") {
```

Comment:
do we accept ssh-style clone URLs, such as `git@github.com:Azure/brigade`? If so should we add a check for that, or should we punt it and document that? Or, perhaps a yes/no prompt for whether we need to attach an SSH key?

Reply:
I'm going to invert the selection, and _not_ ask if the URL is http or https, since those are really the two forms that cannot take an SSH key